### ISSUE TITLE
Spot Rotation

### ISSUE THREAD

**Issue by Atreyus86** [BryanCrotaz/blender-artnet#8]
is it possible to block the refresh of a value that you don't want to be changed?
for example when I have the moving heads in a horizontal position I need to turn the spots on the Z axis to align with the scene on the blender when they are in "locate" mode 127 tilt and 127 pan then with the beam perfectly straight, but when I send again with artnet it resets returning to the zero set by the script and the locate loses its setting.
How can I fix this little problem?

**Comment by BryanCrotaz** [comment]
You’re essentially looking for a fixed offset plus or minus artnet movement?

**Comment by BryanCrotaz** [comment]
You’re essentially rotating the base of a moving spot?

**Comment by Atreyus86** [comment]
essentially you would need an offset to be set directly on the spots where you need to act, and that offset must be available for all three axes.
I have already managed to change the order directly from the script [pan, tilt, 0] or [pan, 0, tilt] but in this particular case where I have to work on a project already done I need to change the degrees of an axis without then it is refreshed by the artnet.

For example:
[pan, tilt, null]

Because there are many moving heads with parent factors, modifiers and more already active

**Comment by BryanCrotaz** [comment]
we could put a checkbox against every DMX channel in the selected fixture. But for a big fixture That would get unwieldy. 

Can you just define a fixture that only has the channels you want to animate?

**Comment by Atreyus86** [comment]
The Issue is that if i patch the spot on the original position, everything works fine. But if i rotate the spot 90° on the X axis because i need the spot rigged in the default position looking in front of the audience (see the attached screenshot) i need to set the offset of the Z axis and leave it there without refresh from the addon. If you try with Blender, when you set the Z axis of the spot, when you switch on the addon, if i don't touch the Z Value, the addon automatically set the value on zero.
Hope to be clear, thanks a lot for the support.
![screenshot1](https://user-images.githubusercontent.com/68188315/89122679-a03ba580-d4c9-11ea-98da-955ce7644bfb.png)
![screenshot2](https://user-images.githubusercontent.com/68188315/89122680-a16cd280-d4c9-11ea-8320-ece07281b3d6.png)

**Comment by BryanCrotaz** [comment]
I think I know what you mean. Please post that second screenshot before and after you patch it so I can see exactly what’s happening and which axis is misbehaving.

**Comment by Atreyus86** [comment]
Before:
![Screenshot (5)](https://user-images.githubusercontent.com/68188315/89126639-8c06a100-d4e7-11ea-9b61-2315aae2445d.png)

After:
![Screenshot (4)](https://user-images.githubusercontent.com/68188315/89126643-96289f80-d4e7-11ea-8563-18117ee492eb.png)

As you can see in the first image i've set up the offset on Z axes without settings the ArtNet values (Pan and Tilt are on X,Y asxes). When i switch on the ArtNet addon, the value of Z become 0 fron the original 65 degrees.
It happens only when i add the ArtNet node in a project that i have created before.

**Comment by Atreyus86** [comment]
And sorry, but when i close the project and after that i re-open the same project, i have to uncheck and recheck the flag on the ArtNet addon and only after that it starts to work again

**Comment by BryanCrotaz** [comment]
Please log the second problem as a separate issue

**Comment by BryanCrotaz** [comment]
So actually you don’t want to rotate the light, you want to rotate the housing. And you want the artnet rotation to be relative to your starting point, not from zero?

Wondering if the lamp body can be set as the parent of the light, and we could have a check box to rotate parent instead of rotating the light?

**Comment by Atreyus86** [comment]
This is a great idea, thank you! After we do this, will the fixture be the same?

**Comment by BryanCrotaz** [comment]
Try the new version 1.4. Pan and tilt are now written to the delta rotation so you can use the rotation transform to set the centre position.

You also have the option to map pan and tilt to parent or grandparent axes.

**Comment by Atreyus86** [comment]
the new version for now is stable and very useful, the option that you added is very helpful, now the workflow is very smooth,more fast and simple.  Thanks a lot. We keep you updated in case of bugs

**Comment by BryanCrotaz** [comment]
If you find it useful commercially I'd be grateful for a Patreon subscription so I can dedicate more time to this.
